src/api/test/functional/request_controller_test.rb

```diff
@@ -857,6 +857,11 @@ def test_create_request_approve_and_review
     assert_response :success
     assert_xml_tag(tag: 'state', attributes: { name: 'review', approver: 'fred' })
 
+    # random guy must not be allowed to drop the request after approval
+    login_adrian
+    post "/request/#{id}?cmd=changestate&newstate=declined"
+    assert_response 403
+
```

Comment:
I think it's better to test this in the model, that is cheaper on the CI, instead of a feature spec, this instantiates the whole stack

Reply:
well, it is anyway instantiate when running this test and we don't need to manually set this situation up a second time there. 
But feel free to create a unit test.